app/models/manageiq/providers/openstack/network_manager/event_target_parser.rb

```diff
@@ -0,0 +1,46 @@
+class ManageIQ::Providers::Openstack::NetworkManager::EventTargetParser
+  attr_reader :ems_event
+
+  # @param ems_event [EmsEvent] EmsEvent object to be parsed to derive an object to be refreshed
+  def initialize(ems_event)
+    @ems_event = ems_event
+  end
+
+  # Parses all targets present in the EmsEvent givin in the initializer
+  # @return [Array] Array of ManagerRefresh::Target objects
+  def parse
+    parse_ems_event_targets(ems_event)
+  end
+
+  private
+
+  # Parses list of ManagerRefresh::Target(s) out of the given EmsEvent
+  #
+  # @param ems_event [EmsEvent] EmsEvent object
+  # @return [Array] Array of ManagerRefresh::Target objects
+  def parse_ems_event_targets(ems_event)
+    target_collection = ManagerRefresh::TargetCollection.new(:manager => ems_event.ext_management_system, :event => ems_event)
```

Comment:
Hm, the separate event catcher for network manager complicates stuff.

I think we need to set here parent_manager:
ManagerRefresh::TargetCollection.new(:manager => ems_event.ext_management_system.parent_manager, :event => ems_event)

Since all of the targeted refresh runs in the cloud manager and the targets are bucketed based on manager. Storage wise, this doesn't matter, since each persistor assigns the right manager for saving. So this is more for which ems and therefore worker should process it.